### PR TITLE
Sample coverage over a gene list

### PR DESCRIPTION
**Pull request by northwestwitch**
### This PR adds | fixes:
- Return coverage over a list of genes (the whole gene, no transcripts) for a sample in the database

### How to test:
- Run the app and query for gene coverage for a sample already saved in the database (should contain a d4 file)

### Expected outcome:
- [x] Gene coverage info should be returned

### Review:
- [x] Code approved by HS
- [x] Tests executed by CR

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: tests/src/chanjo2/endpoints/test_coverage.py [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+from chanjo2.constants import WRONG_COVERAGE_FILE_MSG, WRONG_BED_FILE_MSG
+from chanjo2.demo import gene_panel_file, gene_panel_path
+from chanjo2.models.pydantic_models import (
+    CoverageInterval,
+    Builds,
+)
+from chanjo2.populate_demo import DEMO_SAMPLE
+
+
+def test_d4_interval_coverage_d4_not_found(
```
Comment (northwestwitch) [inline tests/src/chanjo2/endpoints/test_coverage.py:17]: Tests from here are not new, but moved here from test_intervals. I think it makes more sense to have them among the coverage tests.

File: tests/src/chanjo2/endpoints/test_coverage.py [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+    assert response.status_code == status.HTTP_200_OK
+
+    # AND return coverage intervals data
+    result = response.json()
+    for interval in result:
+        assert CoverageInterval(**interval)
+
+
+@pytest.mark.parametrize("build", Builds.get_enum_values())
+def test_samples_gene_coverage_hgnc_symbols(
```
Comment (northwestwitch) [inline tests/src/chanjo2/endpoints/test_coverage.py:166]: Tests from here are new!

**Comment by northwestwitch** [comment]
Query with HGNC symbols: 🆗 

image


Query with HGNC IDs: 🆗 

image


Query with Ensembl IDs: 🆗 

image

File: src/chanjo2/models/pydantic_models.py [hunk 3]
```diff
@@ -129,10 +130,19 @@ class Exon(IntervalBase):
...
-    case_id: Optional[int]
     mean_coverage: float
+
+
+class SampleGeneQuery(BaseModel):
+    sample_name: str
+    build: Builds
+    ensembl_ids: Optional[List[str]]
+    hgnc_ids: Optional[List[int]]
+    hgnc_symbols: Optional[List[str]]
```
Comment (henrikstranneheim) [inline src/chanjo2/models/pydantic_models.py:148]:
```suggestion
    build: Builds
    ensembl_ids: Optional[List[str]]
    hgnc_ids: Optional[List[int]]
    hgnc_symbols: Optional[List[str]]
    sample_name: str
```

File: tests/src/chanjo2/endpoints/test_coverage.py [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+def test_samples_gene_coverage_hgnc_ids(
+    build: str,
+    demo_client: TestClient,
+    endpoints: Type,
+    genomic_ids_per_build: Dict[str, List],
+):
+    """Test the function that returns the coverage over multiple genes of a sampl when a list of HGNC IDs is providede."""
+
+    # GIVING a sample coverage query containing HGNC IDs
+    data = {
```
Comment (henrikstranneheim) [inline tests/src/chanjo2/endpoints/test_coverage.py:201]: Same here

File: tests/src/chanjo2/endpoints/test_coverage.py [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+def test_samples_gene_coverage_ensembl_ids(
+    build: str,
+    demo_client: TestClient,
+    endpoints: Type,
+    genomic_ids_per_build: Dict[str, List],
+):
+    """Test the function that returns the coverage over multiple genes of a sample when a list of Enseml IDs is provided."""
+
+    # GIVING a sample coverage query containing Ensembl IDs
+    data = {
```
Comment (henrikstranneheim) [inline tests/src/chanjo2/endpoints/test_coverage.py:227]: And here

**Review by henrikstranneheim — APPROVED** [review]
Looks great

File: tests/src/chanjo2/endpoints/test_coverage.py (hunk 5 above)
Comment (northwestwitch) [reply tests/src/chanjo2/endpoints/test_coverage.py]: Fixed, thanks!

**Pull request merged**